python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -399,11 +453,25 @@ def run(
                 )
                 raise
 
-        # TODO: get logs from CloudWatch. there are 2 separate streams for stdout and driver stderr to read from
-        # the log group can be found in the response from start_job_run, and the log stream is the job run id
-        # worker logs have log streams like: <job_id>_<worker_id> but we probably don't need to read those
+            response = self._client.get_job_run(JobName=job_name, RunId=run_id)
+            log_group = response["JobRun"]["LogGroupName"]
+            context.log.info(f"Started AWS Glue job {job_name} run: {run_id}")
+
+            response = self._wait_for_job_run_completion(job_name, run_id)
+
+            if response["JobRun"]["JobRunState"] == "FAILED":
+                raise RuntimeError(
+                    f"Glue job {job_name} run {run_id} failed:\n{response['JobRun']['ErrorMessage']}"
+                )
+            else:
+                context.log.info(f"Glue job {job_name} run {run_id} completed successfully")
+
+            if isinstance(self._message_reader, PipesCloudWatchMessageReader):
+                # TODO: receive messages from a background thread in real-time
+                self._message_reader.consume_cloudwatch_logs(
+                    f"{log_group}/output", run_id, start_time=int(start_timestamp)
+                )
```

Comment:
can you expand on this TODO so a person without context understands what this is about?

Reply:
done